test/common/router/header_formatter_test.cc

```diff
@@ -269,6 +269,7 @@ TEST(HeaderParserTest, TestParseInternal) {
       {"%UPSTREAM_METADATA([\"ns\", \n \"key\"])%", {"value"}, {}},
       {"%UPSTREAM_METADATA( \t [ \t \"ns\" \t , \t \"key\" \t ] \t )%", {"value"}, {}},
       {"%START_TIME%", {"2018-04-03T23:06:09.123Z"}, {}},
+      {"%START_TIME(\"testing\")%", {"\"testing\""}, {}},
```

Comment:
Should parseInternal strip the quotes? I would expect that either `%START_TIME(xyz)%` or `%START_TIME("xyz")%` would produce the same output.

Reply:
I can see why you'd say that, but from reading the code it looks to me like a fairly big change to the responsibilities of parseInternal() and hence a reasonably large refactor to shift responsibilities around.  I'd describe what parseInternal currently does as "Find the beginning and end % characters for a range of formats, and give RequestInfoHeaderFormatter the string between those % characters".  This change would imply that parseInternal would either edit that string in a specific way or would give RequestInfoHeaderFormatter a list of tokens based on its parsing.  Is that really what you want?

To put it another way, if we want %START_TIME(xyz)% and %START_TIME("xyz")% to produce the same output, given the current structure of the code the obvious place to do that is in AccessLog::AccessLogFormatParser.  Would you like me to make that change there?

Reply:
Sorry, my browser has been acting weirdly. Thought I responded to this, but it's not here, so I guess not.

Handling the quotes in AccessLogFormatParser will affect %START_TIME% when used in access and upstream logs. I wanted to ask @dio what he thought about all this.

Reply:
@dio: Any thoughts here?

Reply:
Followed the thread above, I think I agree that `%START_TIME(xyz)%` and` %START_TIME("xyz")%` should produce the same output, since this additional enclosing `"` to an arg, as mentioned above, is to state that that arg is a string.

Reply:
Happy to do it, and I'm inclined to do it as part of this PR; if I'm allowing new formats I should probably handle them as they are going to be handled in the future.

Just to give a heads up in case anyone wants to weigh in before I upload the next revision: I think this means I need to modify parsing for %START_TIME% (access log and custom headers) and %DYNAMIC_METADATA% (custom headers only) s.t. if a single string argument is surrounded by quotes it will strip the quotes and use what's inside the quotes as the format string.  

I debated whether I needed to modify %REQ/RESP/TRAILER% parsing and decided that that would be more complex and arguably wouldn't make sense.  Though I could see an argument that quotes should be stripped from the X & Y arguments separately, I couldn't see one for stripping them from the combination.

Reply:
Actually, I'm going to skip DYNAMIC_METADATA, as it's not really a string argument; like REQ/RESP it's several string arguments with delimiting characters.